src/worker.ts

```diff
@@ -0,0 +1,28 @@
+import BN from 'bn.js'
+import workerpool from 'workerpool'
+import { OrderbookFetcher } from './orderbook_fetcher'
+import { sortOrderbookBySerializedPrice } from './utilities'
+import { transitiveOrderbook } from '@gnosis.pm/dex-contracts'
+
+function markets(orderbooks_object: object, base: string, quote: string, hops: number) {
+  const orderbooks = OrderbookFetcher.deserializeOrderbooks(orderbooks_object);
+  const transitive = transitiveOrderbook(orderbooks, base, quote, hops);
+  const result = sortOrderbookBySerializedPrice(transitive);
+  return result;
+}
+
+function estimatedBuyAmount(orderbooks_object: object, base: string, quote: string, hops: number, quoteAmount: string) {
```

Comment:
Maybe the `OrderbookFetcher` is responsible for pushing the new orderbook to each worker instead of serializing and sending everytime?

I think that is a separate improvement for another PR though.

Reply:
When I briefly tested the workers weren't running at all times and thus able to capture state but rather spawned whenever a request came in. Thus I wasn't able to "store" the orderbook inside the worker.

Reply:
I wonder if this is a `workerpool` thing because I was under the impression workers that had message handlers setup would run until explicitly closed. It's been SO LONG since I've done any worker related work so my memory might be off.

Reply:
Iirc when I was looking for libraries some of them used message passing and some had this abstraction of function call (even though internally they still do message passing). Since I picked this library I will stay with it.